src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,53 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final int defaultCircleRadius = 10;
+    private final Color defaultCircleColor = Color.WHITE;
+    private Random randomValue = new Random();
+    private ColorSupplier colorSupp = new ColorSupplier();
+
+    public Figure getRundomFigure() {
+        Figure figure = null;
+        int index = randomValue.nextInt(FigureType.values().length);
+        switch (FigureType.values()[index]) {
+            case CIRCLE:
+                figure = new Circle(
+                        colorSupp.getRandomColor(),
+                        randomValue.nextInt(100));
+                break;
+            case ISOSCELESTRAPEZOID:
+                figure = new IsoscelesTrapezoid(
+                        colorSupp.getRandomColor(),
+                        randomValue.nextInt(100),
+                        randomValue.nextInt(100),
+                        randomValue.nextInt(100));
+                break;
+            case RECTANGLE:
+                figure = new Rectangle(
+                        colorSupp.getRandomColor(),
+                        randomValue.nextInt(100),
+                        randomValue.nextInt(100));
+                break;
+            case RIGHTTRIANGLE:
+                figure = new RightTriangle(
+                        colorSupp.getRandomColor(),
+                        randomValue.nextInt(100),
+                        randomValue.nextInt(100));
+                break;
+            case SQUARE:
+                figure = new Square(
+                        colorSupp.getRandomColor(),

```

Comment:
Lets create 5 private methods getRandomCircle(), getRandomSquare(), ... and call them in switch-case construction, this will increase code readability. (Remember private methods should follow after public as helper ones). And lets make number 100 as constant field